BUILDING.md

```diff
@@ -0,0 +1,78 @@
+# Build containerd from source
+
+This guide is useful if you intend to contribute on containerd. Thanks for your
+effort. Every contribution is very appreciated.
+
+## Gotchas
```

Comment:
I think we can omit this section. Let the users self select on the directions below. No need to provide "dragons be thar" commentary.